lib/amqp_client.js

```diff
@@ -249,47 +266,33 @@ AMQPClient.prototype.createSender = function(address, options) {
 
     var attach = function() {
       var onAttached = function(link) {
-        if (link.name !== linkName) {
-          return;
-        }
-
         debug('sender link attached: ' + linkName);
-        self.removeListener(AMQPClient.LinkAttached, onAttached);
+        link.removeListener(Link.Attached, onAttached);
         self._attached[linkName] = link;
 
         link.on(Link.ErrorReceived, function(err) {
           self.emit(AMQPClient.ErrorReceived, err);
-
-          self._resolveDeferredSenderAttaches(linkName, err);
-          reject(err);
         });
 
         link.on(Link.Detached, function(details) {
           debug('sender link detached: ' + (details ? details.error : 'No details'));
           self._attached[linkName] = undefined;
-
-          self._resolveDeferredSenderAttaches(linkName, details);
-          reject(details);
         });
 
         // return the attached link
         self._resolveDeferredSenderAttaches(linkName, null, link);
         resolve(link);
       };
-      self.on(AMQPClient.LinkAttached, onAttached);
 
       var onMapped = function() {
         self.removeListener(AMQPClient.SessionMapped, onMapped);
-        var linkPolicy = u.deepMerge({
-          options: {
-            name: linkName,
-            source: { address: 'localhost' },
-            target: { address: address }
-          }
-        }, self.policy.senderLink);
-        self._session.attachLink(linkPolicy);
+
+        var link = self._session.attachLink(linkPolicy);
+        link.on(Link.Attached, onAttached);
```

Comment:
I don't necessarily disagree with the original design choice, a lot of the cleanup I'm doing ends up in increased coupling. We should talk about it. But yeah, this definitely solves the outstanding issue for the uuid link names :smile: